protected/humhub/modules/space/widgets/Chooser.php

```diff
@@ -16,6 +16,11 @@
  */
 class Chooser extends Widget
 {
+    /**
+     * Maximum number of spaces to load per request.
+     */
+    const SPACE_BATCH_SIZE = 100;
```

Comment:
This should probably be set even lower, e.g. 25, as there is still one query per space to determine new content count.